components/server/src/analytics.ts

```diff
@@ -46,6 +46,32 @@ export async function trackSignup(user: User, request: Request, analytics: IAnal
     });
 }
 
+export async function trackIDECloseSignal(
```

Comment:
I think it should be inlined in route, there is no need to reuse it somewhere

Reply:
Should we respect the original structure of the code? Make code align to previous is better
 
https://github.com/gitpod-io/gitpod/blob/20fe1e55df48361a3f74a746d5b75fc734d3ae39/components/server/src/user/user-controller.ts#L32

Reply:
I'm not sure what you mean?

The server is built with https://github.com/inversify/InversifyJS for configurations which resolves around classes. If we want to reuse some functions from GitpodSerevr then we should move them to some existing services or to some intermediate which have sense conceptually.

We can also create a new instance of server as it is done here and call it instead: https://github.com/gitpod-io/gitpod/blob/2f0b9462b0451f0974434fa36bd757420ef04e34/components/server/src/user/enforcement-endpoint.ts#L50-L52

Reply:
If we want to use server like this, we will copy-paste needed code in this file 😅

Reply:
I'd like to keep the code

Reply:
Reuse of `enforcement-endpoint` need to refactor `enforcement-endpoint.ts` file too, since it's not that reusable. And have some permission check

Reply:
I don't think we need to touch encofcement endpoint.

We have a user and it should be enough:
```ts
@inject(EnforcementControllerServerFactory) private readonly serverFactory: EnforcementControllerServerFactory;

private createGitpodServer(user: User) {
        const server = this.serverFactory();
        server.initialize(undefined, user, new OwnerResourceGuard(user.id), ClientMetadata.from(user.id), undefined, {});
        return server;
    }
```

we can rename EnforcementControllerServerFactory to GitpodServerFactory?

Reply:
Thank you @akosyakov 🙏